apiserver/apiserver.go

```diff
@@ -332,7 +332,10 @@ func NewServer(ctx context.Context, cfg ServerConfig) (*Server, error) {
 const readyTimeout = time.Second * 30
 
 func newServer(ctx context.Context, cfg ServerConfig) (_ *Server, err error) {
-	controllerDomainServices := cfg.DomainServicesGetter.ServicesForModel(cfg.ControllerModelUUID)
+	controllerDomainServices, err := cfg.DomainServicesGetter.ServicesForModel(ctx, cfg.ControllerModelUUID)
+	if err != nil {
```

Comment:
I know it's necessary, but a bit of a shame. I avoided a change recently so as not to need error returns like this.

Reply:
I know, but luckily the number of locations that we need to do this is small, so although we slightly weaken the API for the services, it doesn't or shouldn't be evident to people using the services day to day. 

If by any stroke of luck we can get rid of the model request when asking for a logger, we can remove the error and it will be back to what it was.